magma/backend/mlir/mem_utils.py

```diff
@@ -0,0 +1,33 @@
+from magma.backend.mlir.hw import hw
+from magma.backend.mlir.mlir import push_block
+from magma.backend.mlir.sv import sv
+
+
+def make_mem_reg(ctx, name, N, T):
+    mem_type = hw.InOutType(hw.ArrayType((N,), T))
+    mem = ctx.new_value(mem_type)
+    sv.RegOp(name=name, results=[mem])
+    return mem
+
+
+def make_mem_read(ctx, target, value, has_enable, name="reg"):
```

Comment:
we may not be perfect throughout on this, but probably prefer not using `"reg"` as a name for a register since it's a reserved keyword in Verilog (MLIR deals with this for us, but might be good practice to avoid anyway).